app/components/ConstructorArgsModal.tsx

```diff
@@ -0,0 +1,74 @@
+import type { AbiConstructor } from "abitype";
+import type React from "react";
+
+interface ConstructorArgsModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  constructorInputs: AbiConstructor["inputs"];
+  constructorArgs: string[];
+  setConstructorArgs: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+const ConstructorArgsModal: React.FC<ConstructorArgsModalProps> = ({
```

Comment:
Can be follow up: I'd actually like to have this in the right hand panel. It's important for all contracts, not just deploying.